modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb

```diff
@@ -16,8 +16,9 @@ class PowerOfAttorneyController < ClaimsApi::BaseFormController
 
         FORM_NUMBER = '2122'
 
-        def submit_form_2122
+        def submit_form_2122 # rubocop:disable Metrics/CyclomaticComplexity
```

Comment:
I think it's an anti-pattern to increase the complexity in controller here. The controller action should be responsible for routing requests and responding to the requests. `#submit_form_2122` is doing a bit more than that it seems. 

I'd suggest a `FormSubmissionService` which could be used to capture the Poa lookup branching logic? 

Reducing cyclomatic complexity and increasing readability would be the goals of that change.

Reply:
Curious: I'm making the assumption that `Form2122` is a form for modifying/assigning power of attorney for a given user - is this accurate?

Reply:
> I'm making the assumption that Form2122 is a form for modifying/assigning power of attorney for a given user - is this accurate?

Yes, that's correct.

Reply:
> I think it's an anti-pattern to increase the complexity in controller here.

Fair point. Ideally I would have been able to simply add source_name to the find_by here and move on, but unfortunately source_name is embedded within source_data which is encrypted in the database, and I couldn't figure out a way to query that value directly :-1: 

I liked the find_by nature of the model so I changed the implementation to this to hopefully help this read a bit better? and reduce cyclomatic complexity at the controller level. Note: rubocop wouldn't let me use the phrasing "find_by_*" because of Rails/DynamicFindBy complaints.
https://github.com/department-of-veterans-affairs/vets-api/commit/fd61ff9f543ad15110d49c05b6cca0a30f88f025